drafts/compound-components.md

```diff
@@ -0,0 +1,252 @@
+  labelActionButton="Delete"
+  isDestructiveAction
+/>
+```
+
+The dialog now looks like this:
+
+![Delete Dialog](https://cdn.hashnode.com/res/hashnode/image/upload/v1709470151176/qBBC62QhH.png?auto=format)
+
+The next requirement is to add a renaming dialog, where you can enter the new file name. The content of the dialog does not only consist of text but also contains an input field. We must be able to pass this to the dialog.
+Another requirement is to convert documents into images. The confirmation dialog should now have three buttons: "Cancel", "Convert to PNG" and "Convert to JPG". We now need to introduce a third button and make it configurable.
+We are only at the beginning of our application and the number of parameters is already starting to get out of hand and there are still many requirements to come.
+
+If we don't want a huge, confusing dialog component that is full of configuration parameters and conditions, we need to take a different approach. One approach that is often chosen is to implement different subtypes of dialogs. Instead of one dialog, you have multiple dialogs for different use cases. One example could be a ConfirmDialog, a dialog that is exclusively used for confirmations of actions.
+Having different dialogs, however, carries the risk that they drift apart and get various looks and feels. There are often so many dialogs that some developers don't know all of them and start introducing new dialogs, even though there is already a dialog for a similar requirement.
+
+This is where the compound component pattern comes to the rescue.
+
+## The Compound Component Pattern
+
+The compound component pattern is a component composition pattern. The aim is to have many small components instead of one large component, all of which require far fewer configuration parameters.
+It always takes four steps to implement a compound component:
+
+1. Create a Context
+2. Create a parent component
+3. Create child components
+4. Add child components as properties to the parent components
+
+All child components can be combined as desired and have a common context.
+The shared context is what makes the pattern so powerful. Not many React developers know and use the pattern. However, the pattern is often used in component libraries as they often require a high level of flexibility. A well-known example is the Headless UI library from tailwind labs.
+In the following chapter we implement a dialog with the compound component pattern. The example will show how the different components are implemented and how the common context is created.
+
+## Implementing a Dialog with the Compound Component Pattern
+
+Let's first see what a dialog looks like, what's common with all of them, and what has to be customizable.
+
+![Parts of a dialog](https://cdn.hashnode.com/res/hashnode/image/upload/v1709464693100/g5q_83GTB.png?auto=format)
+
+- Title: The title has always the same font style but the text has to be customizable
```

Comment:
make the name here bold *Title*, *Close Button*
I think this could help the reader